src/clj/game/core/damage.clj

```diff
@@ -122,29 +123,43 @@
+           (show-wait-prompt state other-player (str (side-str player) " to prevent damage"))
+           (swap! state assoc-in [:prevent :current] type)
+           (show-prompt
+             state player nil (str "Prevent any of the " (- n already-prevented) " " (name type) " damage?") ["Done"]
+             (fn [_] (let [prevent (get-in @state [:damage :damage-prevent type])
+                           damage-prevented (if prevent (- prevent already-prevented) false)]
+                       (if damage-prevented (trigger-event state side :prevented-damage type prevent) nil)
+                       (system-msg state player
+                                   (if damage-prevented (str "prevents " 
+                                                             (if (>= damage-prevented Integer/MAX_VALUE) "all" damage-prevented)
+                                                             " " (name type) " damage") "will not prevent damage"))
+                       (clear-wait-prompt state other-player)
+                       (effect-completed state side eid)))))
+       (effect-completed state side eid)))))
+
 (defn damage
   "Attempts to deal n damage of the given type to the runner. Starts the
   prevention/boosting process and eventually resolves the damage."
+  ([state side eid type n] (damage state side eid type n nil)) 
   ([state side eid type n {:keys [unpreventable card] :as args}]
    (swap! state update-in [:damage :damage-bonus] dissoc type)
    (swap! state update-in [:damage :damage-prevent] dissoc type)
    ;; alert listeners that damage is about to be calculated.
    (trigger-event state side :pre-damage type card n)
-   (let [n (damage-count state side type n args)
-         prevent (get-prevent-list state :runner type)]
-     (if (and (not unpreventable)
-              (cards-can-prevent? state :runner prevent type))
-       ;; runner can prevent the damage.
-       (do (system-msg state :runner "has the option to avoid damage")
-           (show-wait-prompt state :corp "Runner to prevent damage")
-           (swap! state assoc-in [:prevent :current] type)
-           (show-prompt
-             state :runner nil (str "Prevent any of the " n " " (name type) " damage?") ["Done"]
-             (fn [_] (let [prevent (get-in @state [:damage :damage-prevent type])]
-                       (when prevent (trigger-event state side :prevented-damage type prevent))
-                       (system-msg state :runner
-                                   (if prevent (str "prevents " (if (>= prevent Integer/MAX_VALUE) "all" prevent)
-                                                    " " (name type) " damage") "will not prevent damage"))
-                       (clear-wait-prompt state :corp)
-                       (resolve-damage state side eid type (max 0 (- n (or prevent 0))) args)))))
-       (resolve-damage state side eid type n args)))))
+   (let [active-player (get-in @state [:active-player])]
+     (if (not unpreventable)
```

Comment:
this is purely a style thing, but would you mind reversing this? That way, the `if` is checking existence instead of non-existence, and we don't have to look far to see the else branch.
```clojure
(if unpreventable
  (resolve-damage ...)
  (wait-for ...))
```